### PR TITLE
 [installer] Add JWT cookie opts to config WEB-101

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

We want to be able to create/read cookies on both Public API and Server. For that, we should have a shared understanding of the cookie names to look for when extracting the (JWT) session.

* Cookie Name, and opts are specified in `auth.session.cookie` config, used by both server and public-api

## Related Issue(s)

* Depends on https://github.com/gitpod-io/gitpod/pull/17314

## How to test

Unit tests

Preview
1. View
2. Check JWT cookie is set, and the naming
3. Check both server & public-api receive the config in their configmap

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - mp-config-6a7121c1a4</li>
	<li><b>🔗 URL</b> - <a href="https://mp-config-6a7121c1a4.preview.gitpod-dev.com/workspaces" target="_blank">mp-config-6a7121c1a4.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
</ul>

## Build Options:

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

/hold

### CODE REVIEW

File: install/installer/pkg/components/auth/config.go [hunk 1]
```diff
@@ -21,17 +22,40 @@ type Config struct {
...
 func GetConfig(ctx *common.RenderContext) ([]corev1.Volume, []corev1.VolumeMount, Config) {
 	volumes, mounts, pki := getPKI()
+	lifetime := int64((7 * 24 * time.Hour).Seconds())
 	return volumes, mounts, Config{
 		PKI: pki,
 		Session: SessionConfig{
-			LifetimeSeconds: int64((7 * 24 * time.Hour).Seconds()),
+			LifetimeSeconds: lifetime,
 			Issuer:          fmt.Sprintf("https://%s", ctx.Config.Domain),
+			Cookie: CookieConfig{
```
Comment (easyCZ) [inline install/installer/pkg/components/auth/config.go:46]: This becomes the canonical place to set the JWT cookie properties across both components.

Comment (geropl) [reply install/installer/pkg/components/auth/config.go:46]: I understand why we need to pull this config out. But honestly, it's nothing we _should_ configure as we need.

Would love to see a "CAUTION" comment here. Incl. some comments on the config settings, esp. highlighting the security implications. :orange_heart:

Comment (easyCZ) [reply install/installer/pkg/components/auth/config.go:46]: Will do. It's not exposed to the installer, just a way to share the config between the components, which should limit the exposure

Comment (easyCZ) [reply install/installer/pkg/components/auth/config.go:46]: Added

File: install/installer/pkg/components/auth/config_test.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License.AGPL.txt in the project root for license information.
+
+package auth
+
+import "testing"
+
+func TestCookieNameFromDomain(t *testing.T) {
```
Comment (easyCZ) [inline install/installer/pkg/components/auth/config_test.go:9]: Largely generated by ChatGPT, with a prompt to take the JS implementation of this and convert, including write table tests

**Review by geropl — APPROVED** [review]
Works! :tada:

**Comment by easyCZ** [comment]
Thanks for the review!

/unhold

**Pull request merged**